server/src/main/java/keywhiz/KeywhizService.java

```diff
@@ -157,7 +159,8 @@ public void addSecretGenerator(String name, SecretGenerator<?> secretGenerator)
     jersey.register(injector.getInstance(ClientCertificateFilter.class));
 
     logger.debug("Registering response filters");
-    jersey.register(injector.getInstance(SecurityHeadersFilter.class));
+    environment.servlets().addFilter("security-headers-filter", injector.getInstance(SecurityHeadersFilter.class))
+        .addMappingForUrlPatterns(EnumSet.of(DispatcherType.REQUEST), true, "/*");
```

Comment:
Right now, it triggers on 404 but not on 401. I tried adding ERROR and 401s did not get the header.